src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1525,1898 @@ interface MPSMatrixMultiplication {
+		MPSNnImageNode ExportedNodeWithHandle ([NullAllowed] IMPSHandle handle);
+
+		//FIXME: Property name? @property (readwrite, retain, nonatomic) id<MPSHandle> _Nullable handle;
+		[NullAllowed, Export ("handle", ArgumentSemantic.Retain)]
+		IMPSHandle MPSHandle { get; set; }
+
+		[Export ("format", ArgumentSemantic.Assign)]
+		MPSImageFeatureChannelFormat Format { get; set; }
+
+		[Export ("imageAllocator", ArgumentSemantic.Retain)]
+		IMPSImageAllocator ImageAllocator { get; set; }
+
+		[Export ("exportFromGraph")]
+		bool ExportFromGraph { get; set; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(NSObject), Name = "MPSNNStateNode")]
+	[DisableDefaultCtor]
+	interface MPSNnStateNode
+	{
+		//FIXME: MPSHandle? @property (retain, nonatomic) id<MPSHandle> _Nullable handle;
+		[NullAllowed, Export ("handle", ArgumentSemantic.Retain)]
+		IMPSHandle MPSHandle { get; set; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(NSObject), Name = "MPSNNFilterNode")]
+	[DisableDefaultCtor]
+	interface MPSNnFilterNode
+	{
+		[Export ("resultImage")]
+		MPSNnImageNode ResultImage { get; }
+
+		[NullAllowed, Export ("resultState")]
+		MPSNnStateNode ResultState { get; }
+
+		[NullAllowed, Export ("resultStates")]
+		MPSNnStateNode[] ResultStates { get; }
+
+		[Export ("paddingPolicy", ArgumentSemantic.Retain)]
+		IMPSNnPadding PaddingPolicy { get; set; }
+
+		[NullAllowed, Export ("label")]
+		string Label { get; set; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSNnFilterNode), Name = "MPSCNNConvolutionNode")]
+	interface MPSCnnConvolutionNode
+	{
+		[Static]
+		[Export ("nodeWithSource:weights:")]
+		MPSCnnConvolutionNode FromSource (MPSNnImageNode sourceNode, IMPSCnnConvolutionDataSource weights);
+
+		[Export ("initWithSource:weights:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode, IMPSCnnConvolutionDataSource weights);
+
+		[NullAllowed, Export ("convolutionState")]
+		MPSCnnConvolutionStateNode ConvolutionState { get; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (MPSCnnConvolutionNode), Name = "MPSCNNFullyConnectedNode")]
+	interface MPSCnnFullyConnectedNode
+	{
+		[Static]
+		[Export ("nodeWithSource:weights:")]
+		MPSCnnFullyConnectedNode FromSource (MPSNnImageNode sourceNode, IMPSCnnConvolutionDataSource weights);
+
+		[Export ("initWithSource:weights:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode, IMPSCnnConvolutionDataSource weights);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (MPSCnnConvolutionNode), Name = "MPSCNNBinaryConvolutionNode")]
+	interface MPSCnnBinaryConvolutionNode
+	{
+		[Static]
+		[Export ("nodeWithSource:weights:scaleValue:type:flags:")]
+		MPSCnnBinaryConvolutionNode FromSource (MPSNnImageNode sourceNode, IMPSCnnConvolutionDataSource weights, float scaleValue,MPSCnnBinaryConvolutionType type,MPSCnnBinaryConvolutionFlags flags);
```

Comment:
Methods should have verbs: `CreateFromSource`.

Reply:
Same